### PR TITLE
Add smooth corners to `<Card>`

### PR DESCRIPTION
**Pull request by sashachabin**
Add 0.8 smooth corners to `<Card>` with ResizeObserver & [figma-squircle](https://github.com/phamfoo/figma-squircle)

image

### CODE REVIEW

**Comment by sashachabin** [comment]
maybe we need to add debounce 😏

File: client/components/MainPage/Card/useSmoothCorners.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+if ('ResizeObserver' in globalThis) {
+    smoothCornersObserver = new ResizeObserver((entries) => {
+        for (const entry of entries) {
+            setCorners(entry);
+        }
+    });
+}
+
+export function useSmoothCorners(cardRef: MutableRefObject<HTMLElement>) {
```
Comment (hiba9201) [inline client/components/MainPage/Card/useSmoothCorners.ts:30]: it's not always card, right?)

File: client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.module.css [hunk 2]
```diff
@@ -53,15 +53,12 @@
 .MapVehiclesSidebarVehicleInfo {
```
Comment (hiba9201) [inline client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.module.css:53]: Please check if changes in this file are valid?

File: client/components/MainPage/Card/Card.tsx [hunk 3]
```diff
@@ -27,10 +28,16 @@ export function Card({
     headerCaption,
     footerCaption,
 }: CardProps) {
+    const cardRef = useRef<HTMLAnchorElement>(null);
+    const cardTitleRef = useRef<HTMLDivElement>(null);
+    useSmoothCorners(cardRef);
+    useSmoothCorners(cardTitleRef);
```
Comment (hiba9201) [inline client/components/MainPage/Card/Card.tsx:34]: We don't really need smooth corners for all titles? Only for those that have background

**Comment by sashachabin** [comment]
- [ ] Fix all sides (works only top left)

File: client/components/MainPage/Card/Card.tsx (hunk 3 above)
Comment (sashachabin) [reply client/components/MainPage/Card/Card.tsx]: Added condition with bgcolor

File: client/components/MainPage/Card/useSmoothCorners.ts (hunk 1 above)
Comment (sashachabin) [reply client/components/MainPage/Card/useSmoothCorners.ts]: Yep! Added this to Marquee ;)

File: client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.module.css (hunk 2 above)
Comment (sashachabin) [reply client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.module.css]: Committed unnecessary changes. Thank you for your attention!

**Review by hiba9201 — APPROVED** [review]

**Pull request merged**